pkg/core/blockchain.go

```diff
@@ -925,6 +936,23 @@ func (bc *Blockchain) LastBatch() *storage.MemBatch {
 	return bc.lastBatch
 }
 
+// removeOld removes all blocks and transactions from start to stop height inclusively.
+func (bc *Blockchain) removeOld(d dao.DAO, h util.Uint256) error {
+	b, err := bc.GetBlock(h)
+	if err != nil {
+		return err
+	}
+	for _, tx := range b.Transactions {
+		if err := d.DeleteTransaction(tx.Hash()); err != nil {
```

Comment:
I think this logic can be hidden in DAO. And you should drop the block (with transactions) atomically.

Reply:
Atomicity can be done either via a separate memcached or by returning error instead of warning.
First variant seems more viable and can be easier transformed to asynchronous version.

Reply:
Just use a batch.